packages/shorebird_cli/lib/src/commands/preview_command.dart

```diff
@@ -143,12 +147,51 @@ class PreviewCommand extends ShorebirdCommand {
       return ExitCode.software.code;
     }
 
+    final completeRelease = allReleases.firstWhereOrNull(
+      (r) => r.version == releaseVersion,
+    );
+    if (completeRelease != null) {
+      final nonPreviewablePlatforms = completeRelease.activePlatforms.where(
+        (p) => !release.activePlatforms.contains(p),
+      );
+
+      if (nonPreviewablePlatforms.isNotEmpty) {
+        for (final platform in nonPreviewablePlatforms) {
+          final message =
+              '''${platform.displayName} is not previewable and can't be used.''';
+
+          // If the user explicitly specified a platform and it matches a non
+          // previewable platform, we early exit to avoid duplicated warnings/errors.
+          if (results['platform'] == platform.name) {
+            logger.err(message);
+            return ExitCode.software.code;
+          } else {
+            logger.warn(message);
+          }
+        }
+      }
+    }
```

Comment:
It might be good to extract this to a function. It's a bit hard to follow and it's not clear what exactly is being done here without a deeper reading